siliconcompiler/core.py

```diff
@@ -4222,20 +4145,20 @@ def _local_process(self, flow, status, steplist, indexlist):
 
         nodes_to_run = {}
         processes = {}
-        self._prepare_nodes(nodes_to_run, processes, flow, status, steplist, indexlist)
+        self._prepare_nodes(nodes_to_run, processes, flow, status)
         self._launch_nodes(nodes_to_run, processes, status)
-        self._check_nodes_status(flow, status, steplist, indexlist)
+        self._check_nodes_status(flow, status)
 
     ###########################################################################
     def run(self):
         '''
         Executes tasks in a flowgraph.
 
-        The run function sets up tools and launches runs for every index
-        in a step defined by a steplist. The steplist is taken from the schema
-        steplist parameter if defined, otherwise the steplist is defined
-        as the list of steps within the schema flowgraph dictionary. Before
-        starting  the process, tool modules are loaded and setup up for each
+        The run function sets up tools and launches runs for every node
+        in the flowgraph starting with 'from' steps and ending at 'to' steps.
```

Comment:
steps -> nodes

Reply:
You mean 'from' nodes and 'to' nodes?
So far those options are step based so writing nodes here might be confusing to the user.

Reply:
good point